src/macaron/repo_finder/repo_finder.py

```diff
@@ -235,7 +235,9 @@ def get_tags_via_git_remote(repo: str) -> dict[str, str] | None:
     """
     tags = {}
     try:
-        tag_data = git.cmd.Git().ls_remote("--tags", repo)
+        command = git.cmd.Git()
+        command.update_environment(GIT_TERMINAL_PROMPT="")
+        tag_data = command.ls_remote("--tags", repo)
```

Comment:
We already have a [function](https://github.com/oracle/macaron/blob/edfe06eb3f9337f86215b253f40f53ca1a0105c9/src/macaron/environment_variables.py#L10) that manages the environment variables.

It's also better to be consistent with the implementation of [clone_remote_repo](https://github.com/oracle/macaron/blob/edfe06eb3f9337f86215b253f40f53ca1a0105c9/src/macaron/slsa_analyzer/git_url.py#L294) by calling the `git` process. I think the functionality can be implemented in the `git_url.py` module to be reused elsewhere. 

Also, please add a unit test.

Reply:
I wonder what logic should we unit test here 
If we decide to use [`get_patched_env`](https://github.com/oracle/macaron/blob/edfe06eb3f9337f86215b253f40f53ca1a0105c9/src/macaron/environment_variables.py#L10) then there are existing unit tests for env patching [here](https://github.com/oracle/macaron/blob/ec4e1907e0acf213d80ff7c8420e073d9d37494d/tests/test_environment_variables.py#L43)

But if we want to test that calling `git ls-remote` doesn't prompt for user input, then I'm not sure if it is suitable for a unit test because `git ls-remote` accesses external network.